test/game/game_setup_test.exs

```diff
@@ -96,5 +96,18 @@ defmodule GameSetupTest do
       assert players ==
         %Players{player_one: %Player{type: HardComputerPlayer, symbol: "X"}, player_two: %Player{type: HardComputerPlayer, symbol: "O"}}
     end
+
+    test "displays invalid setup validation message via invalid input for player selection, then returns players containing selected players and associated symbols" do
+      user_inputs = ["3", "1", "1"]
+      start_supervised!({@mock_command_line.communicator, user_inputs})
+
+      players = GameSetup.setup_players(@mock_command_line)
+
+      assert CommunicatorMock.get_message_history() =~
+        "Invalid selection, please enter 1 or 2\n"
+
+      assert players ==
+        %Players{player_one: %Player{type: HumanPlayer, symbol: "X"}, player_two: %Player{type: HumanPlayer, symbol: "O"}}
```

Comment:
Just curious why are we returning players if the users are selecting players? Did I misunderstand this test?

Reply:
We return players because both players are selected in this test (via the `GameSetup.setup_players(@mock_command_line)` call with the inputs '1' and '1'), but we also check for the invalid selection prompt that is displayed (as a result of the input '3')

Reply:
Ohhhh I see now. Thanks!